### PR TITLE
Update: Update elide fact adapter

### PR DESCRIPTION
**Pull request by jkusa**
## Description

Update elide fact adapter

## Proposed Changes

- Implement fetchDataForRequest
- Align to elide contracts
- Organize elide fact adapter
- Update tests

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/addon/adapters/elide-facts.ts [hunk 1]
```diff
@@ -62,50 +55,62 @@ export default class ElideFacts extends EmberObject implements NaviFactAdapter {
...
+    while (status === QueryStatus.QUEUED || status === QueryStatus.PROCESSING) {
+      yield timeout(this._pollingInterval);
+      asyncQueryPayload = yield this.fetchAsyncQuery(id);
+      status = asyncQueryPayload?.asyncQuery.edges[0]?.node.status;
+    }
+    return asyncQueryPayload;
+  })
+  fetchDataForRequestTask!: TODO;
+
+  fetchDataForRequest(this: ElideFacts, request: RequestV1, options: RequestOptions): Promise<TODO> {
```
Comment (mattgraham1995) [inline packages/data/addon/adapters/elide-facts.ts:113]: JSDoc

File: packages/data/addon/adapters/elide-facts.ts [hunk 2]
```diff
@@ -62,50 +55,62 @@ export default class ElideFacts extends EmberObject implements NaviFactAdapter {
...
+    let status: QueryStatus = asyncQuery.status;
+
+    while (status === QueryStatus.QUEUED || status === QueryStatus.PROCESSING) {
+      yield timeout(this._pollingInterval);
+      asyncQueryPayload = yield this.fetchAsyncQuery(id);
+      status = asyncQueryPayload?.asyncQuery.edges[0]?.node.status;
+    }
+    return asyncQueryPayload;
+  })
+  fetchDataForRequestTask!: TODO;
```
Comment (mattgraham1995) [inline packages/data/addon/adapters/elide-facts.ts:111]:
```suggestion
  fetchDataForRequestTask!: Promise<AsyncQueryResponse>;
```

File: packages/data/addon/adapters/elide-metadata.ts [hunk 3]
```diff
@@ -38,21 +39,22 @@ export default class GraphQLMetadataAdapter extends EmberObject {
...
-      queryOptions.variables = Object.assign(queryOptions.variables || {}, {
-        ids: [id]
-      });
-    }
+    const queryOptions = {
+      query,
+      variables: {
+        ...(isPresent(id) && { ids: [id] })
+      },
+      ...options
```
Comment (mattgraham1995) [inline packages/data/addon/adapters/elide-metadata.ts:55]: `options` is of type MetadataRequestOptions which also has a `variables` key so we could see some collision there

File: packages/data/addon/adapters/elide-facts.ts (hunk 2 above)
Comment (jkusa) [reply packages/data/addon/adapters/elide-facts.ts:111]: This is a task, this is probably could go on `fetchDataForRequest`

**Pull request merged**